controllers/machine_controller.go

```diff
@@ -173,18 +177,23 @@ func (r *MachineReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cl
 }
 
 func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster, m *clusterv1.Machine) (ctrl.Result, error) {
+	logger := r.Log.WithValues("machine", m.Name, "namespace", m.Namespace)
+	if cluster != nil {
+		logger = logger.WithValues("cluster", cluster.Name)
+	}
+
 	if err := r.isDeleteNodeAllowed(ctx, m); err != nil {
 		switch err {
 		case errNilNodeRef:
-			klog.V(2).Infof("Deleting node is not allowed for machine %q: %v", m.Name, err)
+			logger.Error(err, "Deleting node is not allowed")
```

Comment:
outside the scope of this PR but these should probably return err

Reply:
Returning err means it gets rereconciled with backoff. Not sure that's what we want?

Reply:
ah ok, if the check fails we can try to check again, otherwise it's fine to return nil and not reconcile again 👍